java-processors/src/main/java/com/dataiku/dip/plugins/tradearea/AreaGenerator.java

```diff
@@ -0,0 +1,126 @@
+
+        Input:
+            GeoPoint.Coords center : Coordinates of the center geopoint
+        Output:
+            The rectangular trade area expressed as a WKT polygon (string)
+            example: `POLYGON((long1 lat1,long2 lat2, ...))`
+         */
+        if (this.radius<=0 || this.width<=0 || this.height<=0){
+            logger.info("Detected invalid input parameter. Distance input parameters should be greater than zero.");
+            return null;
+        }
+        GeoPoint.Coords initCoords = null;
+        // Declaration of the final result
+        StringBuilder str = new StringBuilder();
+        str.append("POLYGON((");
+        // Computation of the diagonal angle of the rectangle (Must be in degree and Math.atan is radian)
+        double diagonalAngle = Math.atan(this.height/this.width)*180/Math.PI;
+        double[] angles = {90-diagonalAngle, 90+diagonalAngle, 270-diagonalAngle, 270+diagonalAngle};
+        // Compute the latitude longitude of the four corners and fill the polygon String
+        for (int i = 0; i < angles.length; i++){
+            double angle;
+            angle = angles[i];
+            GeoPoint.Coords tmpCoords;
+            tmpCoords = GeoUtils.computeDestinationPoint(center.latitude, center.longitude, angle, this.radius);
+            str.append(tmpCoords.longitude + " " + tmpCoords.latitude + ",");
+            if (i==0){
+                initCoords = tmpCoords;
+            }
+        }
+        // Close the polygon by adding initial geopoint
+        str.append(initCoords.longitude + " " + initCoords.latitude);
+        str.append("))");
+        return str.toString();
+    }
+}
+
+class CircleAreaGenerator extends AreaGenerator {
+    /*
+    Generates a circular area centered on an input geopoint coordinates.
+    The parameter radius are expected to be in kilometers.
+    If using miles in the processor, those distances must be converted to miles before.
+     */
+
+    double radius;
+
+    public CircleAreaGenerator(double radius){
+        this.radius = radius;
+    }
+
+    public String generateArea(GeoPoint.Coords center) {
+        /*
+        Generate an almost circular area approximated with 12 points
+
+        Input:
+            GeoPoint.Coords center: The center of the trade area as a geospatial instance coordinates
+        Output:
+            The circular trade area expressed as a WKT POLYGON (String)
+            example: `POLYGON((long1 lat1,long2 lat2, ...))`
+         */
+        if (this.radius<=0){
+            logger.info("Detected invalid radius as input. Radius parameter should be greater than zero.");
+            return null;
+        }
+        GeoPoint.Coords initCoords = null;
+        StringBuilder str = new StringBuilder();
+        str.append("POLYGON((");
+        // Compute the points on circle, angle step is set to 30 degrees as there are 12 points
+        for (int i = 0; i < 12; i++){
```

Comment:
Create a constant for 12:
```suggestion
class CircleAreaGenerator extends AreaGenerator {
       static final int NB_OF_EDGES = 360;
       ....
       for (int i = 0; i < NB_OF_EDGES; i++){
```

Reply:
Ok